bin/ruby-build

```diff
@@ -244,24 +244,33 @@ compute_md5() {
   fi
 }
 
+declare -a HAS_CHECKSUM_SUPPORT
+has_checksum_support() {
+  local checksum_command="$1"
+  return ${HAS_CHECKSUM_SUPPORT[$checksum_command]:=$(echo test | "$checksum_command" >/dev/null; echo $?)}
+}
+
 verify_checksum() {
-  # If there's no SHA2 support, return success
-  [ -n "$HAS_SHA2_SUPPORT" ] || return 0
-  local checksum_command="compute_sha2"
+  local checksum_command
+  local filename="$1"
+  local expected_checksum="$(echo "$2" | tr [A-Z] [a-z])"
 
   # If the specified filename doesn't exist, return success
-  local filename="$1"
   [ -e "$filename" ] || return 0
 
-  # If there's no expected checksum, return success
-  local expected_checksum=`echo "$2" | tr [A-Z] [a-z]`
-  [ -n "$expected_checksum" ] || return 0
+  case "${#expected_checksum}" in
+  # If expected checksum is empty, return success
+  0) return 0 ;;
+  # MD5
+  32) checksum_command="compute_md5" ;;
+  # SHA2 256
+  64) checksum_command="compute_sha2" ;;
+  # unknown checksum algorithm, return failure
+  *) return 1 ;;
```

Comment:
Can you add a test for the case of checksum being of unsupported length? Should we also output a friendly error message about it if we're going to abort the build process?